crates/notary/server/tee/notary-server-sgx.md

```diff
@@ -0,0 +1,68 @@
+# TLSNotary Notary Server for Intel SGX
+
+This package contains the **SGX-enabled version of the TLSNotary Notary Server**, built with **Gramine** to run securely on Intel SGX hardware. Follow the instructions below to install dependencies, verify the integrity of the enclave, and launch the server.
+
+## Install Intel SGX Drivers & Runtime Dependencies
+
+Ensure your system has the required Intel SGX components installed.
+
+```sh
+wget https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key
+cat intel-sgx-deb.key | sudo tee /etc/apt/keyrings/intel-sgx-keyring.asc > /dev/null
+
+# Add the repository to your sources:
+echo 'deb [signed-by=/etc/apt/keyrings/intel-sgx-keyring.asc arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu noble main' | sudo tee /etc/apt/sources.list.d/intel-sgx.list
+
+sudo apt-get update
+sudo apt-get install libsgx-epid libsgx-quote-ex libsgx-dcap-ql -y
+```
+
+For more details, refer to the official **[Intel SGX Installation Guide](https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/Intel_SGX_SW_Installation_Guide_for_Linux.pdf).**
+
+## Verify the Enclave Signature (`MR_ENCLAVE`, `MR_SIGNER`)
+
+Before running the Notary Server, verify that the provided **enclave signature (`.sig`)** matches the expected **`MR_ENCLAVE` and `MR_SIGNER`** values.
+
+Run the following command inside a **Gramine Docker container** to inspect the enclave's signature:
```

Comment:
seems like the following expects that one already has the github artefact zipfile (that contain the notary server binary, sig file) downloaded?

if so, is this entire verification process mainly to ensure that the downloaded artefact is valid and untampered?

Reply:
Yes. It is not really necessary to run these checks, the SGX will refuse to run the binary if anything is off.

I'll document how to verify with a local build later.